File: CPP(C++)/Encapsulation/Encapsulation.cpp
#include<iostream>
using namespace std;
// Create a Class Or Blue_Print Of Your Objecct
class Company {
    private:
        // Attributes of your class or varibles in pr. mode
        int EmpId;
        string EmpName;
        string ExpJobDesc;
        int Age;
        float Salary;
    public:
        // set private data types using functions.
        void setEmpID(int id){
            EmpId = id;
        }
        void setEmpName(string name){
            EmpName = name;
        }
        void setEmpJobDesc(string desc){
            ExpJobDesc = desc;
        }
                void setAge(int age){
            Age = age;
        }
                void setSalary(float salary){
            Salary = salary;
        }
        // wappiing the data and function together
        // we can also use contructors too.
        void getDetails(){
            cout<<"Emp. Name :- "<<EmpName<<"\nEmp. Job :- "<<ExpJobDesc<<"\nEmp. Age :- "<<Age<<"\nSalary :- "<<Salary<<endl;
        }
};
int main(){
    Company object;
    object.setEmpID(2966);
    object.setEmpName("Abhay Singh");
    object.setEmpJobDesc("Student");
    object.setAge(19);
    object.setSalary(50000);  
    object.getDetails();
}

File: Python(.py)/Encapsulation/Encapsulation.py

class Company:
    # this a constructor used to define your class attributes.
    def __init__(self):
        self.EmpId = 0
        self.EmpName = ""
        self.EmpJob = ""
        self.Age = 0
    # create Our Method (Special Method) To initialize our Attributes of class 
    def __init__(self,id,name,job,age):
        self.EmpId = id
        self.EmpName = name
        self.EmpJob = job
        self.Age = age
    # crete a Method. to get details of Employee
    def getDetails(self):
        print(f"Emp Id :- {self.EmpId}\nEmp. Name :- {self.EmpName}\nEmp. Job :- {self.EmpJob}\nEmp Age :- {self.Age}\n")

# main Function In Python 
if __name__ == "__main__":
    # object Creation 
    obj = Company(2966,"Abhay Singh","Student",19)
    obj2 = Company(2967,"Radhey Singh","Student",18)
    # getDetails Method called 
    obj.getDetails()
    obj2.getDetails()
